src/main/java/org/example/servletsHomework/servlet/LoginServlet.java

```diff
@@ -0,0 +1,46 @@
+package org.example.servletsHomework.servlet;
+
+
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.example.servletsHomework.model.User;
+import org.example.servletsHomework.storage.TokensAndUserStorage;
+
+import java.io.IOException;
+
+public class LoginServlet extends HttpServlet {
+
+    private final TokensAndUserStorage tokensAndUserStorage;
+
+    public LoginServlet(TokensAndUserStorage tokensAndUserStorage) {
+        this.tokensAndUserStorage = tokensAndUserStorage;
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+
+        String oldToken = req.getHeader("token");
+
+        if (tokensAndUserStorage.ifTokenExists(oldToken)) {
+
+            String newToken = tokensAndUserStorage.generateToken();
+            User oldUser = tokensAndUserStorage.removeToken(oldToken);
+            User newUser = new User(
```

Comment:
why you create new user when old token exists ? I  thought only tokens can be changed but user remain the same